p24/src/refs.rs

```diff
@@ -0,0 +1,27 @@
+pub fn f1(s: &mut (u32, u32), flag: bool) -> &mut u32 {
+    match flag {
+        false => &mut s.0,
+        true => &mut s.1,
+    }
+}
+
+pub fn f2(s: &mut [u32], length: usize) -> &mut u32 {
+    &mut s[length]
+}
+
+pub fn f3(s: &mut [u32], length: usize) -> &mut u32 {
+    &mut s[s.len() - length - 1]
+}
+
+pub fn f4(s: &mut [u32]) -> [&[u32]; 4] {
```

Comment:
This function can accept a shared reference (i.e. &[u32]). Also, the implemented algorithm is not "fair", e.g. for slice with length equal to 5 it will return slices with lengths equal to 1, 1, 1, and 4.